nbqa/__main__.py

```diff
@@ -201,8 +137,16 @@ def _replace_temp_python_file_references_in_out_err(
     err
         Stderr with temporary directory replaced by current working directory.
     """
-    out, err = _replace_full_path_out_err(out, err, temp_python_file, notebook)
-    out = _replace_relative_path_out_err(out, notebook)
+    out = (
+        out.replace(f"{tempdirname}{os.sep}", "")
+        .replace(f"{str(Path(tempdirname).resolve())}{os.sep}", "")
+        .replace("   .py", ".ipynb")
+    )
+    err = (
+        err.replace(f"{tempdirname}{os.sep}", "")
+        .replace(f"{str(Path(tempdirname).resolve())}{os.sep}", "")
+        .replace("   .py", ".ipynb")
+    )
```

Comment:
IMHO all that string replacing should stay function, which than can be unittested.

Reply:
But with only two calls, this is mostly aesthetics, so highly subjective 😝

Reply:
True :smile: IMO with just two simple calls, this is likely OK, but if we run into bugs down the line then extracting a function from here, with unit tests, might be in order

Reply:
@girip11 not sure I understand, sorry.

On `master`, I get:
```python
$ nbqa flake8 tests/data/notebook_for_testing.ipynb 
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:1:1: F401 'os' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:3:1: F401 'glob' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:5:1: F401 'nbqa' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_2:2:1: E302 expected 2 blank lines, found 0
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_2:19:9: W291 trailing whitespace
```

while on this branch I get

```python
$ nbqa flake8 tests/data/notebook_for_testing.ipynb 
tests/data/notebook_for_testing.ipynb:cell_1:1:1: F401 'os' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:3:1: F401 'glob' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:5:1: F401 'nbqa' imported but unused
tests/data/notebook_for_testing.ipynb:cell_2:2:1: E302 expected 2 blank lines, found 0
tests/data/notebook_for_testing.ipynb:cell_2:19:9: W291 trailing whitespace
```

If I just run `flake8`, I get

```
$ flake8 nbqa/replace_source.py --max-line-length=80
nbqa/replace_source.py:2:81: E501 line too long (85 > 80 characters)
nbqa/replace_source.py:43:81: E501 line too long (81 > 80 characters)
nbqa/replace_source.py:51:81: E501 line too long (85 > 80 characters)
nbqa/replace_source.py:70:81: E501 line too long (86 > 80 characters)
```

so wouldn't the change proposed in this PR would make the output more consistent with flake8's? (similarly for other tools)

Reply:
Sorry. My bad. Ignore the comment.

Reply:
Your comment was valid though, as with that change we had

```python
$ nbqa flake8 /home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb
tests/data/notebook_for_testing.ipynb:cell_1:1:1: F401 'os' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:3:1: F401 'glob' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:5:1: F401 'nbqa' imported but unused
tests/data/notebook_for_testing.ipynb:cell_2:2:1: E302 expected 2 blank lines, found 0
tests/data/notebook_for_testing.ipynb:cell_2:19:9: W291 trailing whitespace
```

With the change I've just pushed, instead, we have:

```python
$ nbqa flake8 /home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:1:1: F401 'os' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:3:1: F401 'glob' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_1:5:1: F401 'nbqa' imported but unused
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_2:2:1: E302 expected 2 blank lines, found 0
/home/marco/nbQA-dev/tests/data/notebook_for_testing.ipynb:cell_2:19:9: W291 trailing whitespace

$ nbqa flake8 tests/data/notebook_for_testing.ipynb
tests/data/notebook_for_testing.ipynb:cell_1:1:1: F401 'os' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:3:1: F401 'glob' imported but unused
tests/data/notebook_for_testing.ipynb:cell_1:5:1: F401 'nbqa' imported but unused
tests/data/notebook_for_testing.ipynb:cell_2:2:1: E302 expected 2 blank lines, found 0
tests/data/notebook_for_testing.ipynb:cell_2:19:9: W291 trailing whitespace
```

which seems more like what we're after

I'll write a failing test (should've done that first :flushed:) and'll push again later